airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py

```diff
@@ -280,21 +278,19 @@ def __init__(
         self,
         stream: AbstractFileBasedStream,
         message_repository: MessageRepository,
-        sync_mode: SyncMode,
         cursor_field: Optional[List[str]],
         state: Optional[MutableMapping[str, Any]],
         cursor: "AbstractConcurrentFileBasedCursor",
     ):
         self._stream = stream
         self._message_repository = message_repository
-        self._sync_mode = sync_mode
         self._cursor_field = cursor_field
         self._state = state
         self._cursor = cursor
 
     def generate(self) -> Iterable[FileBasedStreamPartition]:
         pending_partitions = []
-        for _slice in self._stream.stream_slices(sync_mode=self._sync_mode, cursor_field=self._cursor_field, stream_state=self._state):
+        for _slice in self._stream.stream_slices(sync_mode=SyncMode.incremental, cursor_field=self._cursor_field, stream_state=self._state):
```

Comment:
Something that we could do if we're not convinced we will always have a cursor for this reason, but don't want to let the user define that from their end, is add handling to the `ensure_at_least_one_state_emitted` of `FileBasedConcurrentCursor` to check for empty states and emit our new no cursor sentinel in that case.

Reply:
>Interesting thing that goes against this hypothesis: although we require a cursor for instantiation, [acually getting the cursor field](https://github.com/airbytehq/airbyte/pull/37021/files#diff-66098b071cfd5f979ca57af71ec11e579eeffa3523741f7bb15ea4d2da80a6c0R59) can still return None and has handling [here](https://github.com/airbytehq/airbyte/pull/37021/files#diff-66098b071cfd5f979ca57af71ec11e579eeffa3523741f7bb15ea4d2da80a6c0R73). Im not sure what happens in this case, but something feels off here

Maybe you already saw this, but in case not and it's useful - `get_cursor_field_from_stream` is not specific to file-based streams (it was broken off as a helper function to be used by the file-based CDK). I believe that all file-based sources have a cursor field, so if we wanted to get it off of the stream in a different way that doesn't require us to handle the `None` I think that would make sense.

Reply:
The file based stream defines its own cursor field [here](https://github.com/airbytehq/airbyte/blob/694605251383beecd6e56cdf2fada5081557ad66/airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/cursor/default_file_based_cursor.py#L19)

We don't really use the cursor_field returned by  `get_cursor_field_from_stream`, we just pass it down into the stream_slices method